### PR TITLE
Add ESP BLE GATT server support

### LINKED ISSUE #5926: Add BLE  GATT server support to espressif port
ESP32, ESP32-S3 and ESP32-C3 can all do BLE.

Let's add `_bleio` support on top of NimBLE that the IDF uses for BLE.

Once it's added, we can then support the BLE workflow on these ESP chips.

### PR DESCRIPTION
**Pull request by tannewt**
Enable BLE where we can. Switch 4MB, non-USB board default partitioning over to a single 2MB firmware bank. Old boards override this setting to keep the same behavior.

This also adds alpha support for the ESP32-C2 (aka ESP8684).

Fixes #5926 and #7170

### CODE REVIEW

**Comment by tannewt** [comment]
Seems to work on S3. Connecting hangs on C6. C2 runs out of memory with wifi on. C3 is untested. ESP32 crashes with watchdog timeout.

File: ports/espressif/common-hal/_bleio/Adapter.c [hunk 1]
```diff
@@ -201,6 +221,7 @@ static int _scan_event(struct ble_gap_event *event, void *scan_results_in) {
             disc->data,
             disc->length_data);
     } else {
+        #if MYNEWT_VAL(BLE_EXT_ADV)
```
Comment (dhalbert) [inline ports/espressif/common-hal/_bleio/Adapter.c:224]: What is the use case for not always having ` MYNEWT_VAL(BLE_EXT_ADV)` turned on?

File: ports/espressif/Makefile [hunk 2]
```diff
@@ -469,11 +486,15 @@ update-all-sdkconfigs: $(BUILD)/esp-idf/config/sdkconfig.h
 update-board-sdkconfig: $(BUILD)/esp-idf/config/sdkconfig.h
 	python tools/update_sdkconfig.py --board=$(BOARD) --debug=$(DEBUG)
 
-BINARY_WIFI_BLOBS = libcore.a libespnow.a libmesh.a libnet80211.a libpp.a libsmartconfig.a libwapi.a
```
Comment (dhalbert) [inline ports/espressif/Makefile:472]: Does `libmesh.a` have anything to do with ESP-NOW support? What is `libwapi.a` for ?
(I just don't know. I tried a websearch but didn't find an answer.)

**Review by dhalbert — CHANGES_REQUESTED** [review]
Is this lib/tinyusb submodule change a regression? I thought it was bumped forward in another PR.

Just some other questions. I did not test. I will put the "this and that don't work" in the release notes. When this is merged, could you add issues addressing the ports that don't work? Thanks.

**Comment by tannewt** [comment]
> Is this lib/tinyusb submodule change a regression? I thought it was bumped forward in another PR.

This bumps it further add definitions for the C2.

> Just some other questions. I did not test. I will put the "this and that don't work" in the release notes. When this is merged, could you add issues addressing the ports that don't work? Thanks.

I'm not sure what you mean by this. I hope it all works but I expect folks to find issues.

File: ports/espressif/common-hal/_bleio/Adapter.c (hunk 1 above)
Comment (tannewt) [reply ports/espressif/common-hal/_bleio/Adapter.c:224]: The ESP32 doesn't support BLE 5 so it doesn't support extended advertisements.

File: ports/espressif/Makefile (hunk 2 above)
Comment (tannewt) [reply ports/espressif/Makefile:472]: I don't actually know either. I just know the H2 doesn't have them available.

**Comment by dhalbert** [comment]
> I'm not sure what you mean by this. I hope it all works but I expect folks to find issues.

I meant what you said in the first post:

> Connecting hangs on C6. C2 runs out of memory with wifi on. C3 is untested. ESP32 crashes with watchdog timeout.

C2 is not surprising. So C6 and ESP32 needs some fixing, and C3 needs to be tested.

**Review by dhalbert — APPROVED** [review]

**Comment by tannewt** [comment]
Ah! I think C6 and ESP32 are fixed.RAM may still be an issue but we'll see.

**Pull request merged**